src/main/java/graphql/nadel/DefinitionRegistry.java

```diff
@@ -42,17 +44,30 @@ public SchemaDefinition getSchemaDefinition() {
         return (SchemaDefinition) definitionsByClass.get(SchemaDefinition.class).get(0);
     }
 
+    public Map<OperationType, ObjectTypeDefinition> getOperationTypes(){
+        Map<OperationType, ObjectTypeDefinition > opsTypesMap = new HashMap<>();
+
+        Arrays.stream(OperationType.values()).forEach(opsType ->
+                opsTypesMap.put(opsType, getOpsType(opsType.getOpsType(), opsType.getDisplayName())));
+
+       return opsTypesMap;
+    }
+
     public ObjectTypeDefinition getQueryType() {
+        return getOpsType(OperationType.QUERY.getOpsType(), OperationType.QUERY.getDisplayName());
+    }
+
+    private ObjectTypeDefinition getOpsType(String typeName, String typeDisplay ) {
```

Comment:
naming nitpick but typeName sounds like a "graphql type".   "operationTypeName / operationDisplayName" might be better names